courses/management/commands/create_enrollment.py

```diff
@@ -0,0 +1,113 @@
+
+    help = "Creates an enrollment for a course run"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--user",
+            type=str,
+            help="The id, email, or username of the User",
+            required=True,
+        )
+        parser.add_argument(
+            "--run",
+            type=str,
+            help="The 'courseware_id' value for the CourseRun",
+            required=True,
+        )
+        parser.add_argument(
+            "--code", type=str, help="The enrollment code for the course", required=True
+        )
+        super().add_arguments(parser)
+
+    def handle(self, *args, **options):
+        """Handle command execution"""
+
+        user = fetch_user(options["user"])
+
+        run = CourseRun.objects.filter(courseware_id=options["run"]).first()
+        if run is None:
+            raise CommandError(
+                "Could not find course run with courseware_id={}".format(options["run"])
+            )
+
+        product = Product.objects.filter(
+            courseruns__courseware_id=options["run"]
+        ).first()
+        if product is None:
+            raise CommandError(
+                "No product found for that course with courseware_id={}".format(
+                    options["run"]
+                )
+            )
+
+        user_enrollment = CourseRunEnrollment.objects.filter(user=user, run=run)
+        if user_enrollment:
+            raise CommandError(
+                "User={} is already enrolled in courseware_id={}".format(
+                    options["user"], options["run"]
+                )
+            )
+
+        coupon = Coupon.objects.filter(coupon_code=options["code"])
+        if not coupon:
+            raise CommandError(
+                "That enrollment code {} does not exist".format(options["code"])
+            )
+
+        # Check if the coupon is valid for the product
+        coupon_version = best_coupon_for_product(product, user, code=options["code"])
+        if coupon_version is None:
+            raise CommandError(
+                {"coupons": "Enrollment Code {} is invalid".format(options["code"])}
+            )
+
+        try:
+            enroll_in_edx_course_runs(user, [run])
+            edx_request_success = True
+        except (EdxApiEnrollErrorException, UnknownEdxApiEnrollException):
+            self.stdout.write(
+                self.style.WARNING(
+                    "Failed to enroll in edx course {}".format(options["run"])
+                )
+            )
+            edx_request_success = False
+
+        enrollment, created = CourseRunEnrollment.objects.get_or_create(
+            user=user,
+            run=run,
+            order=None,
+            defaults=dict(edx_enrolled=edx_request_success),
+        )
+        if not created and not enrollment.active:
+            raise CommandError(
+                "Failed to enroll in MIT xPRO course {}".format(options["run"])
+            )
```

Comment:
If the enrollment exists but is inactive this check would just fail the command but not attempt to activate the enrollment, which probably is the correct way to go.